internal/server/server_test.go

```diff
@@ -1,47 +1,96 @@
 package server
 
 import (
+	"log"
 	"net/http"
 	"net/http/httptest"
 	"os"
 	"testing"
 
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/gorilla/sessions"
 	"github.com/stretchr/testify/assert"
 )
 
-var testHandler http.Handler
+var server *Server
 
 func TestMain(m *testing.M) {
 	var err error
-	testHandler, err = New()
+	server, err = New(
+		WithMockOIDCNewProvider(MockOIDCNewProvider),
+	)
 	if err != nil {
+		log.Printf("server.New failed, error: %v", err)
 		os.Exit(1)
 	}
 
+	server.oauth2Config = &MockOauth2Config{}
+	server.verifier = oidc.NewVerifier("", &MockKeySet{}, &oidc.Config{ClientID: server.kubectlClientID})
+
 	os.Exit(m.Run())
 }
 
 func TestHandleRoot(t *testing.T) {
 	t.Parallel()
 
 	w := httptest.NewRecorder()
-	testHandler.ServeHTTP(w, httptest.NewRequest("GET", "/", nil))
+	server.ServeHTTP(w, httptest.NewRequest("GET", "/", nil))
 	assert.Equal(t, http.StatusOK, w.Code)
 }
 
 func TestHandleLoginGet(t *testing.T) {
 	t.Parallel()
 
 	w := httptest.NewRecorder()
-	testHandler.ServeHTTP(w, httptest.NewRequest("GET", "/login", nil))
+	server.ServeHTTP(w, httptest.NewRequest("GET", "/login", nil))
 	assert.Equal(t, http.StatusMethodNotAllowed, w.Code)
 }
 
 func TestHandleLoginPost(t *testing.T) {
 	t.Parallel()
 
 	w := httptest.NewRecorder()
-	testHandler.ServeHTTP(w, httptest.NewRequest("POST", "/login", nil))
-	assert.Equal(t, http.StatusOK, w.Code)
+	server.ServeHTTP(w, httptest.NewRequest("POST", "/login", nil))
+	assert.Equal(t, http.StatusSeeOther, w.Code)
 	assert.NotEmpty(t, w.Result().Cookies()[0])
 }
+
+func TestHandleCallbackGet(t *testing.T) {
```

Comment:
This could be refactored to use table tests.

Reply:
Fixed